processing/src/main/java/io/druid/segment/data/GenericIndexed.java

```diff
@@ -407,35 +216,56 @@ public T get(int index)
   @Override
   public int indexOf(T value)
   {
-    return bufferIndexed.indexOf(value);
+    return indexOf(this, value);
+  }
+
+  private int indexOf(Indexed<T> indexed, T value)
```

Comment:
JIT won't necessarily figure out the right `indexed`.  At least, if there are multiple concrete types of `Indexed` being passed, JIT might not be able to actually pick just a single one.  Which will mean that every call to `get` will end up with an indirection.  The previous implementation only has one indirection (the one for `indexOf`) and after that it's all local calls to the specific instance.  I agree that this isn't the primary bottleneck, but am more wanting to make sure you understand that the previous implementation is better for JIT.